nixos/modules/services/monitoring/dcgm.nix

```diff
@@ -0,0 +1,208 @@
+{
+  config,
+  pkgs,
+  lib,
+  ...
+}:
+with lib; let
+  cfg = config.services.dcgm;
+in {
+  options.services.dcgm = {
+    enable = mkOption {
+      type = types.bool;
+      default = false;
+      description = lib.mdDoc ''
+        NVIDIA Data Center GPU Manager (DCGM).
+        Runs as a background process to manage GPUs on the node.
+        Provides interface to address queries from DCGMI tool.
+      '';
+    };
+
+    port = mkOption {
+      type = types.port;
+      default = 5555;
+      description = lib.mdDoc ''
+        Specify the port for the Host Engine.
+      '';
+    };
+
+    domainSocket = mkOption {
+      type = types.nullOr types.path;
+      default = null;
+      description = lib.mdDoc ''
+        Specify the Unix domain socket path for host engine.
+        No TCP listening port is opened when this option is specified.
+      '';
+    };
+
+    bindInterface = mkOption {
+      type = types.str;
+      default = "127.0.0.1";
+      description = lib.mdDoc ''
+        Specify the IP address of the network interface that the host engine should listen on.
+
+        ALL = bind to all interfaces.
+
+        NOTE: This must be an IP address. Do not use a hostname.
```

Comment:
There's a special syntax for notes that will be picked up by the documentation

```
::: {.note}
  This is a note
:::
```

See more here: https://nixos.org/manual/nixpkgs/stable/#sec-contributing-markup